internal/signalfx-agent/pkg/monitors/subproc/core.go

```diff
@@ -192,8 +192,8 @@ func makePipes() (*messageReadWriter, io.ReadCloser, io.WriteCloser, error) {
 // subprocess.  MontiorCore.ConfigureInSubproc handles configuring the
 // subprocess monitor and collecting the result.
 type MessageHandler interface {
-	ProcessMessages(context.Context, MessageReceiver)
-	HandleLogMessage(io.Reader) error
+	ProcessMessages(context context.Context, messageReceiver MessageReceiver)
```

Comment:
nit: idiomatic golang
```suggestion
	ProcessMessages(ctx context.Context, messageReceiver MessageReceiver)
```

Reply:
Update lines 195 and 196 to use ctx and r.